build/openresty/patches/ngx_lua-0.10.25_01-dyn_upstream_keepalive.patch

```diff
@@ -1,8 +1,33 @@
+diff --git a/bundle/nginx-1.21.4/src/http/ngx_http_upstream.c b/bundle/nginx-1.21.4/src/http/ngx_http_upstream.c
+index b07e564..9e25905 100644
+--- a/bundle/nginx-1.21.4/src/http/ngx_http_upstream.c
++++ b/bundle/nginx-1.21.4/src/http/ngx_http_upstream.c
+@@ -4304,6 +4304,7 @@ ngx_http_upstream_next(ngx_http_request_t *r, ngx_http_upstream_t *u,
+     if (u->peer.cached && ft_type == NGX_HTTP_UPSTREAM_FT_ERROR) {
+         /* TODO: inform balancer instead */
+         u->peer.tries++;
++        u->peer.notify(&u->peer, u->peer.data, NGX_HTTP_UPSTREAM_NOFITY_CACHED_CONNECTION_ERROR);
```

Comment:
Do we need to check if `u->peer.notify` is null  on the nginx path unless we have already initialized u->peer.notify to make sure it will not be null

I found in stream module, it also has notify field in u->peer, and it always check it:

```
$ grep 'peer.notify' -R src/
src//stream/ngx_stream_proxy_module.c:    if (u->peer.notify) {
src//stream/ngx_stream_proxy_module.c:        u->peer.notify(&u->peer, u->peer.data,

```

Reply:
Yes, I think we should check it as nginx does.